src/Processors/Pipe.h

```diff
@@ -107,6 +107,8 @@ class Pipe
     void setLeafLimits(const SizeLimits & leaf_limits);
     void setQuota(const std::shared_ptr<const EnabledQuota> & quota);
 
+    void setMaxParallelStreams(size_t num_threads) { max_parallel_streams = num_threads; }
```

Comment:
It is not a good idea to set `max_parallel_streams` externally. It may brake internal invariant (see comment)